dor/providers/resource_provider.py

```diff
@@ -0,0 +1,51 @@
+from pathlib import Path
+
+from dor.providers.file_provider import FileProvider
+from dor.providers.models import PackageResource
+from dor.providers.parsers import DescriptorFileParser, PreservationEventFileParser
+
+
+class DescriptorFileNotFound(Exception):
+    pass
+
+
+class ResourceProvider:
+
+    def __init__(self, file_provider: FileProvider, resource_path: Path):
+        self.file_provider: FileProvider = file_provider
+        self.resource_path: Path = resource_path 
+
+    def get_descriptor_path(self) -> Path:
+        file_paths = list((self.resource_path / "descriptor").iterdir())
+        if len(file_paths) < 0:
+            raise DescriptorFileNotFound()
+        return file_paths[0]
+
+    def get_resource(self) -> PackageResource:
+        descriptor_file_parser = DescriptorFileParser(
+            self.get_descriptor_path(),
+            self.file_provider
+        )
+
+        pres_file_paths = descriptor_file_parser.get_preservation_file_paths()
+        event_file_paths = [
```

Comment:
@ssciolla Matching on the filename doesn't feel like an improvement over `try...pass`. If looking for `PREMIS:event` is not an acceptable option, we should find a different `USE` to distinguish `PREMIS:object` (which matches the definition of `PROVENANCE`) and `PREMIS:event`. _Looks through thesaurus._ Actually, why couldn't we use `USE=PREMIS:object` and `USE=PREMIS:event`? _Words._

Reply:
Yeah, I was also not sure about matching on the file name. It just seems like we should be able to programmatically distinguish between these files beyond whether they have an `event` inside of them. What you're suggesting makes sense to me.

Reply:
I mean, looking for an element _is_ programatically distinguishing between the files ¯\_(ツ)_/¯ Not any different than looking at the headers to distinguish between JPEG and PNG. Since we can make up the `USE` vocabulary, let's refine it. (`REPRESENTATION` and `EVENT` would also work, if we didn't want to leak namespaces everywhere.)

Reply:
This is taken care of, we changed events to "USE" `EVENT`.